fs/attr.c

```diff
@@ -244,7 +309,19 @@ int notify_change(struct dentry * dentry, struct iattr * attr, struct inode **de
 			return -EPERM;
 
 		if (!inode_owner_or_capable(inode)) {
+#if CONFIG_TRUENAS
+			if (IS_NFSV4ACL(inode)) {
+				error = inode_permission(inode, MAY_WRITE);
```

Comment:
So, the assumption is, if you've got `MAY_WRITE`, you can not only modify the data but attributes as well. However, the inverse is not necessarily true, correct ?

Reply:
Correct. Issue here is actually the spec states that user having either ACE4_WRITE_DATA or ACE4_WRITE_ATTRIBUTES allows user to set time to _current_ server time. Doesn't allow arbitrary times. That's why it's permitted here, but not above. I should probably make this checks more explicit though (error == EACCES).